git-worktree/src/lib.rs

```diff
@@ -1 +1,62 @@
 #![forbid(unsafe_code, rust_2018_idioms)]
+//! Git Worktree
+
+use anyhow::Result;
+use git_hash::oid;
+use git_index::{entry::Mode, State};
+use git_object::Data;
+use std::fs::create_dir_all;
+use std::path::Path;
+use std::str;
+
+mod options;
+
+pub use options::Options;
+
+/// Copy index to `path`
+pub fn copy_index<P, Find>(state: &State, path: P, mut find: Find, opts: Options) -> Result<()>
+where
+    P: AsRef<Path>,
+    Find: for<'a> FnMut(&oid, &'a mut Vec<u8>) -> Option<Data<'a>>,
+{
+    let path = path.as_ref();
+    let entries = state.entries();
+    let mut buf = Vec::new();
+    for entry in entries {
+        let dest = path.join(entry.path(state).to_string());
+        create_dir_all(dest.parent().expect("Path has no parent"))?;
```

Comment:
I think most of the code already written optimizes for reading the `expect()` statement, so here it would be something like `file path has a directory`.